pygmt/src/directional_rose.py

```diff
@@ -0,0 +1,102 @@
+"""
+directional_rose - Add a map directional rose.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+from pygmt._typing import AnchorCode
+from pygmt.alias import Alias, AliasSystem
+from pygmt.clib import Session
+from pygmt.helpers import build_arg_list
+
+
+def directional_rose(
+    self,
+    position,
+    position_type: Literal["user", "justify", "mirror", "normalize", "plot"] = "user",
+    width: float | str | None = None,
+    justify: AnchorCode | None = None,
+    offset: Sequence[float | str] | None = None,
+    label: Sequence[str] | bool = False,
+    fancy: Literal[1, 2, 3] | bool = False,
+):
+    r"""
+    Add a directional rose to the map.
+
+    Parameters
+    ----------
+    position/position_type
+        Location of the rose. The actual meaning of this parameter depends on the
+        ``position_type`` parameter.
+
+        - ``position_type="user"``: *position* is given as (x, y) in user coordinates.
+        - ``position_type="normalize"``: *position* is given as (nx, ny) in normalized
+          coordinates, where (0, 0) is the lower-left corner and (1, 1) is the
+          upper-right corner of the map.
+        - ``position_type="plot"``: *position* is given as (x, y) in plot coordinates.
+        - ``position_type="justify"``: *position* is given as a two-character
+          justification code, meaning the anchor point of the rose is inside the map
+          bounding box.
+        - ``position_type="mirror"``: *position* is given as a two-character
+          justification code, but the rose is outside the map bounding box.
```

Comment:
Hm. I am not sure if "justify" and "mirror" are suitable / understandable terms here, especially "mirror" appears a bit strange for me. Is this orientated on some other implementaiton or documentation?

Reply:
GMT uses long names like `mapcoords`,`inside`,`outside`,`boxcoords`,`plotcoords`
https://github.com/GenericMappingTools/gmt/blob/98a9b39c1a94ff23dba8dd89fca9fae367432303/src/longopt/psbasemap_inc.h#L30

but `gmtlogo` uses `user|map`,`justify`,`mirror`,`normalize`,`plot`
https://github.com/GenericMappingTools/gmt/blob/98a9b39c1a94ff23dba8dd89fca9fae367432303/src/longopt/gmtlogo_inc.h#L29

Reply:
All the parameters/arguments need more discussions.